SDK/AppCenter/Microsoft.AppCenter.Windows.Shared/Storage/StorageAdapter.cs

```diff
@@ -151,6 +153,83 @@ private List<object[]> ExecuteSelectionSqlQuery(string query, IList<object> args
             }
         }
 
+        private long GetMaxPageCount()
+        {
+            return GetPragmaValue("max_page_count");
+        }
+
+        private long GetPageCount()
+        {
+            return GetPragmaValue("page_count");
+        }
+
+        private long GetPageSize()
+        {
+            return GetPragmaValue("page_size");
+        }
+
+        private long GetPragmaValue(string valueName)
+        {
+            var result = ExecuteSelectionSqlQuery($"PRAGMA {valueName};");
+            var count = (long)(result.FirstOrDefault()?.FirstOrDefault() ?? 0L);
+            return count;
+        }
+
+        public bool SetMaxStorageSize(long sizeInBytes)
+        {
+            bool success;
+            var db = _db ?? throw new StorageException("The database wasn't initialized.");
+
+            // Check the current number of pages in the database to determine whether the requested size will shrink the database.
+            long currentPageCount = GetPageCount();
+            long pageSize = GetPageSize();
+            AppCenterLog.Info(AppCenterLog.LogTag, $"Found {currentPageCount} pages in the database.");
+            long requestedMaxPageCount = Convert.ToBoolean(sizeInBytes % pageSize) ? sizeInBytes / pageSize + 1 : sizeInBytes / pageSize;
+
+            if (currentPageCount > requestedMaxPageCount)
+            {
+                AppCenterLog.Warn(AppCenterLog.LogTag, $"Cannot change database size to {sizeInBytes} bytes as it would cause a loss of data. " +
+                    "Maximum database size will not be changed.");
+                success = false;
+            }
+            else
+            {
+
+                // Attempt to set the limit and check the page count to make sure the given limit works.
+                int result = raw.sqlite3_exec(_db, $"PRAGMA max_page_count = {requestedMaxPageCount};");
+                if (result != raw.SQLITE_OK)
+                {
+                    AppCenterLog.Error(AppCenterLog.LogTag, $"Could not change maximum database size to {sizeInBytes} bytes. SQLite error code: {result}.");
+                    success = false;
+                }
+                else
+                {
+                    long currentMaxPageCount = GetMaxPageCount();
```

Comment:
Use var if possible (like in other places in this project)

Reply:
Fixed